spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/groovy/org/springframework/cloud/contract/verifier/plugin/ContractsCopyTask.groovy

```diff
@@ -16,115 +16,111 @@
 		}
 	}
 
-	@CompileDynamic
-	private File outputContractsFolder(String root) {
-		File outputContractsFolder = outputFolder(root, "contracts")
-		ext.contractsDslDir = outputContractsFolder
-		return outputContractsFolder
-	}
-
-	@CompileDynamic
-	private String root(ContractVerifierConfigProperties props) {
-		String root = OutputFolderBuilder.buildRootPath(project)
-		ext.contractVerifierConfigProperties = props
-		return root
-	}
-
-	@CompileDynamic
-	private void convertBackedUpDslsToYaml(String root, File file, String antPattern, String slashSeparatedAntPattern, ContractVerifierConfigProperties props, File outputContractsFolder) {
-		File originalContracts = outputFolder(root, ORIGINAL_PATH)
-		copy(file, antPattern, slashSeparatedAntPattern, props, originalContracts)
+	private void convertBackedUpDslsToYaml(File file, String antPattern, String slashSeparatedAntPattern, File outputContractsFolder, boolean excludeBuildFolders) {
+		sync(file, antPattern, slashSeparatedAntPattern, excludeBuildFolders, backupContractsFolder.get().asFile)
 		ToYamlConverter.replaceContractWithYaml(outputContractsFolder)
-		project.logger.
-				info("Replaced DSL files with their YAML representation at [" + ext.contractsDslDir + "]")
+		logger.info("Replaced DSL files with their YAML representation at [" + outputContractsFolder + "]")
 	}
 
-	@CompileDynamic
-	protected WorkResult copy(File file, String antPattern, String slashSeparatedAntPattern, props, File outputContractsFolder) {
-		return project.copy {
-			from(file)
-			// by default group id is slash separated...
-			include(antPattern)
-			// ...we also want to allow dot separation
-			include(slashSeparatedAntPattern)
-			if (props.isExcludeBuildFolders()) {
-				exclude "**/target/**", "**/build/**", "**/.mvn/**", "**/.gradle/**"
+	protected WorkResult sync(File file, String antPattern, String slashSeparatedAntPattern, boolean excludeBuildFolders, File outputContractsFolder) {
+		// TODO: Is there any better way to make it statically compiled, avoiding explicit creation of new Action?
+		// sync will remove files from target if they are removed from source. So using it here instead of copy:
+		return project.sync(new Action<CopySpec>() {
+			@Override
+			void execute(final CopySpec spec) {
+				spec.with {
+					from(file)
+					// by default group id is slash separated...
+					include(antPattern)
+					// ...we also want to allow dot separation
+					include(slashSeparatedAntPattern)
+					if (excludeBuildFolders) {
+						exclude "**/target/**", "**/build/**", "**/.mvn/**", "**/.gradle/**"
+					}
+					into(outputContractsFolder)
+				}
 			}
-			into(outputContractsFolder)
-		}
+		})
 	}
 
-	@CompileStatic
-	private File outputFolder(String root, String suffix) {
-		return getExtension().stubsOutputDir != null ?
-				project.file("${getExtension().stubsOutputDir}/${root}/${suffix}") :
+	static File createTaskOutput(String root, File stubsOutputDir, Project project, String suffix) {
```

Comment:
This appears unused.

Reply:
It's used in the `SpringCloudContractVerifierGradlePlugin`